test/e2e/clusterview_test.go

```diff
@@ -0,0 +1,198 @@
+					"verbs": []interface{}{
+						"list",
+					},
+				},
+				{
+					"apiGroups": []interface{}{
+						"cluster.open-cluster-management.io",
+					},
+					"resources": []interface{}{
+						"managedclusters",
+					},
+					"resourceNames": clusters,
+					"verbs": []interface{}{
+						"list", "get",
+					},
+				},
+			},
+		},
+	}
+}
+
+func generateClusterRoleBinding(name, clusterRole, user string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "rbac.authorization.k8s.io/v1",
+			"kind":       "ClusterRoleBinding",
+			"metadata": map[string]interface{}{
+				"name": name,
+			},
+			"roleRef": map[string]interface{}{
+				"apiGroup": "rbac.authorization.k8s.io",
+				"kind":     "ClusterRole",
+				"name":     clusterRole,
+			},
+			"subjects": []map[string]interface{}{
+				{
+					"kind":     "User",
+					"apiGroup": "rbac.authorization.k8s.io",
+					"name":     user,
+				},
+			},
+		},
+	}
+}
+
+func createClusterRoleAndRoleBinding() {
+	clusterRole := generateClusterRoleWithClusters(clusterRoleName, []string{"cluster1"})
+
+	_, err := util.ApplyClusterResource(dynamicClient, util.ClusterRoleGVR, clusterRole)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	clusterRoleBinding := generateClusterRoleBinding(clusterRoleBindingName, clusterRoleName, userName)
+	_, err = util.ApplyClusterResource(dynamicClient, util.ClusterRoleBindingGVR, clusterRoleBinding)
+}
+
+func validateClusterView(UserDynamicClient dynamic.Interface, ViewGVR, resourceGVR schema.GroupVersionResource, expectedNames []string) error {
+	resourceList, err := util.ListResource(UserDynamicClient, ViewGVR, "", "")
+	if err != nil {
+		return fmt.Errorf("validateClusterView: failed to List Resource %v", err)
+	}
+
+	if len(resourceList) != len(expectedNames) {
+		return fmt.Errorf("validateClusterView: reources count %v != expected count %v", len(resourceList), len(expectedNames))
+	}
+	for _, item := range resourceList {
+		name, _, err := unstructured.NestedString(item.Object, "metadata", "name")
+		if err != nil {
+			return fmt.Errorf("validateClusterView: failed to get resource name %v", err)
+		}
+		exist := false
+		for _, expectedName := range expectedNames {
+			if name == expectedName {
+				exist = true
+				break
+			}
+		}
+		if !exist {
+			return fmt.Errorf("validateClusterView: resource %v is not in expected resource list %v", name, expectedNames)
+		}
+
+		rsExisted, err := util.HasResource(dynamicClient, resourceGVR, "", name)
+		if err != nil {
+			return fmt.Errorf("validateClusterView: failed to get resource %v. err:%v", name, err)
+		}
+		if !rsExisted {
+			return fmt.Errorf("validateClusterView: no resource %v", name)
+		}
+	}
+
+	return nil
+}
+
+var _ = ginkgo.Describe("Testing ClusterView", func() {
+	ginkgo.It("should get the managedClusters.clusterview successfully", func() {
+		// create clusterRole and clusterRoleBinding for user
```

Comment:
i think we should also have cases for creating/deleting clusterrole

Reply:
added

Reply:
watch case is added too